### ISSUE TITLE
Multipage features disabled when TIFF comes with well-established .FAX extension

### ISSUE THREAD

**Issue by trickshooter** [d2phap/ImageGlass#1091]
### System information:
 - Windows OS version: Win10 20H2 64bit
 - ImageGlass version: 8.2.6.6

### To Reproduce
1. Open a multipage Tiff CCITT Fax Group 4 file in ImageGlass - file has extension *.fax (!)
2. Result: None of the features for multipage handling are available, neither keyboard shortcuts nor any of the menu functions
3. Rename the very same file to extension *.tiff or *.tif
4. Result: All multipage features are available as desired

### Actual behavior:
No multipage support for a TiffG4 file when it comes with a file extension of *.fax

### Expected behavior:
Full multipage support for any TiffG4file - no matter under what file extension flag it might be sailing.

### Screenshots:
-none-

### Additional context:
-none-

**Comment by d2phap** [comment]
ImageGlass uses file extension to for viewing feature, I will add FAX to the next release

**Comment by trickshooter** [comment]
Great to hear that - I really can't wait to see it running!

But wait, there is possibly more to it ..... because it is **equally vital** for me to be able to PRINT this very *.fax file as well!

As I could see from your printing dialogue, you are using the Windows built-in specialized "photo printing" functions rather than a windows standard printing dialog or some self-written type of dialog.

Thus, you might likely come across a follow-up problem here:

Please, also check whether printing is available for (single and multipage TIFF G4) files with extension *.fax - since there is a decade(!) old and still unsolved problem with "Windows Photo Viewer" and "Windows Photo Gallery" only being able to view, but (as of today = Win10 64bit 21H1) still **UNABLE TO PRINT** any TIFF files that do NOT specifically carry the extension *.tif or *.tiff (I googled for:  error printing fax tif "windows can't find this picture" ).

https://answers.microsoft.com/en-us/windows/forum/windows_7-pictures/cannot-print-pictures-error-windows-cant-find-this/5597403b-9605-4be2-8d5e-db76fcfcf6d0

Feel free to comment!

Best regards
Christoph

**Comment by d2phap** [comment]
Add multipages support for TIFF, FAX.
You can try the latest build: https://imageglass.org/moon

**Comment by trickshooter** [comment]
This refers to ImageGlass_Moon_8.2.8.9_x64

OK, multipage support for extension *.tif, *.fax and *.tiff now works flawlessly as desired regarding the pure **viewing** functionality (only). Thank you so much for implementing these additional file extensions! BUT (unfortunately)....

(for all the below cases we're always talking about a binarily identical TiffG4 file that simply gets renamed to different file extensions)

As I already lined out in my previous post above, there _might_ be a **printing** issue with *.fax files as well - and it is as I already feared:

In case of **printing** the respective multipage tif file, there is still a severe issue:
*.tif as wells as *.tiff are handled correctly (the printing dialog correctly offers all pages to be printed or selected or cycled through (in the printing preview), BUT...

**_...NOT so with *.fax files!_** Printing always only offers one single page (the current = last viewed) to be printed without an option to scroll through pages (as if the file is a single page file).

**The possibility of printing all pages of the \*.fax file is mandatory for the issue to be resolved!**

@d2phap: Possible solution / suggestion:
Behind the scenes, first copy *.fax file to %temp%\\\*.tif before passing it to the Windows standard photo print routine, because multipage printing of *.tif and *.tiff is handled correctly by Windows - whereas handling of *.fax files is **NOT.**

Greetz,
Christoph

**Comment by d2phap** [comment]
@trickshooter do you have a sample file of FAX extension?

**Comment by trickshooter** [comment]
Sure -  voilà!   -->    [test-files.zip](https://github.com/d2phap/ImageGlass/files/7478795/test-files.zip)

The test files have been created by using Word365, then printed to a TiffG4-creating batch print queue, that stems from www.pdf24.org (which is basically a locally running ghostscript wrapper application). So basically the test file is just a simple multipage TiffG4 file, created by ghostscript and finally just **renamed** from *.tif to *.fax

So, the two contained test files are binarily identical. Still, Windows is only able to print all pages of the *.tif (also if named *.tiff), while failing on the identical *.fax and only offering 1 single page at all to be printed.

BTW, you'll be facing the same, similar or related problems, when trying to view /print these kind of files (i.e. *.fax) using the on-board or built-in Windows software from Microsoft like "Windows Live Photo Gallery", "Microsoft Photos App", "Windows Photo Viewer", etc.

Keep up the good work, my friend!

Best regards,
Christoph

**Comment by d2phap** [comment]
Thanks @trickshooter 
This should be fixed in IG v8.3
https://imageglass.org/news/announcing-imageglass-kobe-8-3-73

**Comment by trickshooter** [comment]
Thanks for the good work @d2phap! I have tested what you implemented and it works fine so far.

BUT ... before I was able to test the new feautures, I had to manually add *.fax extension to the list of the 77 supported file formats. Without that, it was not possible to print the file or even to "see" it when browsing (with PgUp/PgDown) through a directory (containing 4 sample files: fax, tif, tiff, jpg). ImageGlass was only presenting "3/3" files (instead of 4) and it did NOT even show the .fax file when cycling though the directory. Only double-clicking the .fax (or mouse dragging it into ImageGlass window) ultimately showed it and it disappeared as soon as I started to cycle through the containing dir.

**Please add .fax to the default list of supported file formats** to make it work for everybody without having to add the file extension .fax in settings _manually_ after installation.

In file "igconfig.xml"
```
(...)
<Item key="AllFormats" value=" (...);*.fax;(...)" />
(...)
```

Thank you!

Best regards,
Christoph
